maildump_runner/__main__.py

```diff
@@ -37,7 +37,10 @@ def terminate_server(sig, frame):
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('--smtp-ip', default='127.0.0.1', metavar='IP', help='SMTP ip (default: 127.0.0.1)')
-    parser.add_argument('--smtp-port', default=1025, type=int, metavar='PORT', help='SMTP port (deault: 1025)')
+    parser.add_argument('--smtp-port', default=1025, type=int, metavar='PORT', help='SMTP port (default: 1025)')
+    parser.add_argument('--smtp-auth', action='store_true', help='Enable SMTP authorization')
```

Comment:
Since maildump is only meant for development/testing, what about always allowing SMTP AUTH, and simply accepting any username/password; also accepting mails both with and without authentication. Or accept everything is SMTP AUTH is enabled, and verify username/password only if specified.

Reply:
OK, I can't add comment to your next question, so I answer here:
> I'm curious, what client are you using maildump with that requires smtp auth?

I am using UserEngage for transactional emails for both, real and test/qa envs. UserEngage doesn't accept SMTP without authorization, and as I wrote as an answer to other question:

> it's only for testing - I need to catch all transactional emails we are sending on test/qa env (i don't want to send any email to real world just because one of testers/developers used an email like aaa@gmail.com ;) )

Reply:
ok, exposing it to the world is probably not a good idea - no guarantees for security, since it was never meant to be exposed to the public. Anyway, with an IP restriction is probably not that bad

What about reusing a htpasswd-style file for smtp auth? We already have this for the web interface:

    --htpasswd HTPASSWD   Apache-style htpasswd file

so doing the same for smtp wouldn't be too hard, e.g.

--smtp-auth HTPASSWD

Reply:
Yep, I agree with note about security.
About htpasswd - I didn't think about this. It's interesting idea, because we can use more then one credentials set. I will look at this soon :)

Reply:
@ThiefMaster I've changed this for using HTPASSWD style file for SMTP authorization, also added short message about security of this solution.